### DISCUSSION TITLE
Can't build LearnOpenGL projects

### DISCUSSION THREAD

**Discussion by DaniaNouro** [MuhammadSulaiman001/opengl-lab#10] [Q&A] [answered]
استاذ لما وصل لتعليمة 
cmake --build .
عم يعطيني الف ايرور 
متل هيك
 
cmake —build . 
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
Warning: corrupt .drectve at end of def file
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0x7ff): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0x924): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0x94a): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0x970): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0xcbf): undefined reference to `_RTC_CheckStackVars'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0xcd2): undefined reference to `__security_check_cookie'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0xe16): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0xe9a): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0xf1b): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0xf3f): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0xfcf): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.text$mn+0xff2): more undefined references to `__imp__wassert' follow
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.xdata+0x24): undefined reference to `__GSHandlerCheck'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.rtc$IMZ+0x0): undefined reference to `_RTC_InitBase'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/window.obj):(.rtc$TMZ+0x0): undefined reference to `_RTC_Shutdown'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/input.obj):(.text$mn+0x4a): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/input.obj):(.text$mn+0x15f): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/input.obj):(.text$mn+0x5da): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/input.obj):(.text$mn+0x6ca): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/input.obj):(.text$mn+0x7b0): undefined reference to `__imp__wassert'
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/input.obj):(.text$mn+0x8c2): more undefined references to `__imp__wassert' follow
D:/learnopen_Gl/LearnOpenGL/lib/glfw3.lib(glfw.dir/Debug/input.obj):(.text$mn+0x1e51): undefined reference to `__report_rangecheckfailure'
يعني المشكلة من وين ماعم اعرف

**Comment by MuhammadSulaiman001** [comment, marked as answer]
Hi, There is a [similar issue](https://github.com/JoeyDeVries/LearnOpenGL/issues/350) at LearnOpenGL  repo, and [related question](https://stackoverflow.com/questions/24897138/what-does-warning-corrupt-drectve-at-end-of-def-file-mean)s on SO.

I think cmake is not using MSVC compiler (or using an outdated one, so, please uninstall any other version of visual studio other than 2022, especially 2012!).

You can make sure cmake is using the VS2022 compiler using cmake gui.

First, create a folder on desktop and give it a name (ex. cmake-output), then, from start menu, search for `cmake gui`

![image](https://github.com/MuhammadSulaiman001/opengl-lab/assets/36730723/faa2494f-ad87-4f90-bf3f-ffa4b1d004c6)

Open it, then specify the paths for `where is your source code:` and `where to build the binaries:`, In your case, the source code path is `D:/learnopen_Gl/LearnOpenGL/`, and the binaries path should point to `cmake-output` folder (that is created in the first step).

Next, press on `configure`, and make sure to choose Visual Studio 2022

![image](https://github.com/MuhammadSulaiman001/opengl-lab/assets/36730723/a3e64995-7151-4d1d-b1d1-8bab89d46e10)

Click Finish, then Generate, and that's it.. The projects and .sln files should be generated on `cmake-output` directory.
